frontend/Plugin.js

```diff
@@ -0,0 +1,191 @@
+  }
+
+  /**
+   * Initiate and setup the SDK
+   */
+  initPlugin() {
+    window.emosTrackVersion = 2;
+    window.emosClientKey = this.clientKey;
+
+    // Load the econda SDK (emos2.js)
+    initSDK();
+
+    this.register.pageview(({ page }) => {
+      // Some page should be ignored here
+      if (this.ignoredPageviews.includes(page.shopgateUrl)) {
+        return;
+      }
+
+      const emospro = getEmosproForUrl(page.merchantUrl);
+
+      // Add additional data for some pages
+      switch (page.shopgateUrl) {
+        case 'cart':
+        case 'cart_empty':
+          emospro.orderProcess = '1_Warenkorb';
+          break;
+        case 'login':
+          emospro.orderProcess = '2_Anmeldung';
+          break;
+        default:
+      }
+
+      this.send(emospro);
+    });
+
+    this.register.viewContent((data, { product }) => {
+      const emospro = {
+        ...getEmosproForUid(product.uid),
+        ec_Event: [getProductEventData(product)],
+      };
+
+      this.send(emospro);
+    });
+
+    this.register.variantSelected((data, { baseProduct, variant }) => {
+      const emospro = {
+        type: 'event',
+        ec_Event: [getProductEventData(baseProduct, variant)],
+      };
+
+      this.send(emospro, true);
+    });
+
+    this.register.addToCart((data, { products }, scope, state) => {
+      const baseProduct = getCurrentBaseProductFormatted(state);
+      const [variant] = products;
+
+      const emospro = {
+        ...getEmosproForUid(baseProduct.uid),
+        ec_Event: [{
+          ...getProductEventData(baseProduct, variant),
+          type: 'c_add',
+        }],
+      };
+
+      this.send(emospro);
+    });
+
+    this.register.loginSuccess((user) => {
+      const emospro = {
+        ...getEmosproForUrl('login'),
+        login: [CryptoJs.MD5(user.id).toString(), 0],
+      };
+
+      this.send(emospro);
+    });
+
+    this.register.loginFailed(() => {
+      const emospro = {
+        ...getEmosproForUrl('login'),
+        login: ['0', 1],
+      };
+
+      this.send(emospro);
+    });
+
+    let previousSearch = null;
+
+    this.register.search((data) => {
+      /**
+       * Track a search result page only once per searchTerm.
+       * Ignores pageviews where only the sorting changed.
+       */
+      if (previousSearch === data.query) {
+        return;
+      }
+
+      const emospro = {
+        ...getEmosproForUrl(`search?s=${data.query}`),
+        search: [data.query, data.hits],
+      };
+
+      this.send(emospro);
+
+      previousSearch = data.query;
+    });
+
+    this.register.purchase((data, { order }, scope, state) => {
+      const userId = state.user.data ? state.user.data.id : '';
+
+      const emospro = {
+        ...getEmosproForUrl(`checkout_success/${order.number}`),
+        orderProcess: '7_Bestaetigung',
+        billing: [
+          order.number,
+          CryptoJs.MD5(userId).toString(),
+          'NULL', // TODO: LOCATION. Not available at the moment
```

Comment:
Is this a todo still?

Reply:
it's not a todo for now, because the location is optional and there is no way at the moment to get it